### ISSUE TITLE
Setup `jsx-a11y/alt-text` ESLint rule properly

### ISSUE THREAD

**Issue by HonkingGoose** [Belco90/octoclairvoyant#227]
## Describe the bug

We have enabled the [`jsx-a11y/alt-text`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/alt-text.md) ESLint rule, but it didn't report the error since we are using a custom `Image` component for rendering the `img` tag.

## Give the steps to reproduce

1. Create `img` tag without a `alt` property
2. Run ESLint: doesn't report the missing `alt` property
3. Use browser dev tools to inspect `img`, see missing `alt text`

## Have you thought of a possible solution?

Setup `jsx-a11y/alt-text` ESLint rule properly so we catch problems with our alt text in other parts of the site as well.

## Do you want to help fix the bug?

I don't know how to setup ESLint correctly so that it catches this error properly.

**Comment by Belco90** [comment]
Damn! I fixed this without realizing I did it directly pushing to `main` branch 🤦‍♂️ 

This is the commit: https://github.com/Belco90/octoclairvoyant/commit/6827723ce53a3858990b5f5a635082314668ab06

It turns out `jsx-a11y/alt-text` was already setup properly by `eslint-config-next`. However, I didn't setup this config properly, so it's fixed now. You can see it complains if I remove the `alt` prop:
<img width="631" alt="imagen" src="https://user-images.githubusercontent.com/2677072/125195774-73eee400-e257-11eb-8329-2434f5a7803f.png">

Apart from that, I replaced the `xo` base config with `kentcdodds` one because `xo` 1) doesn't seem properly maintained (or some subconfigs at least) and 2) it was colliding with TS config in a really strange way. I know `kentcdodds` config (it has nice defaults) so I'm switching to one then.

I also set ESLint max warnings to 0 since some configs are enabling some rules as warnings, so we don't allow pushing code with warning problems either.

If you are happy with those changes, feel free to close this issue. If you prefer me to revert everything and put it in a proper PR, let me know and I can do it later.

**Comment by HonkingGoose** [comment]
> It turns out `jsx-a11y/alt-text` was already setup properly by `eslint-config-next`. However, I didn't setup this config properly, so it's fixed now. You can see it complains if I remove the `alt` prop:
> <img alt="imagen" width="631" src="https://user-images.githubusercontent.com/2677072/125195774-73eee400-e257-11eb-8329-2434f5a7803f.png">

Cool! That's what we wanted! 😄 

> Apart from that, I replaced the `xo` base config with `kentcdodds` one because `xo` 1) doesn't seem properly maintained (or some subconfigs at least) and 2) it was colliding with TS config in a really strange way. I have used `kentcdodds` config before and it has nice defaults, so I'm switching to one then.

Let's use the package/config we know works properly then. 😉 

> I also set ESLint max warnings to 0 since some configs are enabling some rules as warnings, so we don't allow pushing code with warning problems either.

Agree! We should try to fix stuff right away, instead of leaving it till later.

> If you are happy with those changes, feel free to close this issue. If you prefer me to revert everything and put it in a proper PR, let me know and I can do it later.

I'm happy with these changes! Thank you for fixing the config! ❤️ 

---

Issue closed by commit 6827723
